java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -561,34 +563,74 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_listSortRows(JNIEnv *env,
 }
 
 JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_ColumnView_stringSplit(JNIEnv *env, jclass,
-                                                                        jlong column_view,
-                                                                        jlong delimiter_ptr,
-                                                                        jint max_split) {
-  JNI_NULL_CHECK(env, column_view, "column is null", 0);
-  JNI_NULL_CHECK(env, delimiter_ptr, "string scalar delimiter is null", 0);
+                                                                        jlong input_handle,
+                                                                        jstring pattern, jint limit,
+                                                                        jboolean split_by_regex) {
+  JNI_NULL_CHECK(env, input_handle, "input_handle is null", 0);
+
+  if (limit == 0 || limit == 1) {
```

Comment:
Even if it is different why do we care. The point of this is to provide an API to cudf. Not provide something that is compatible with existing java APIs. So long as it does not crash/leak things, then we should let this through.

Reply:
Okay then I'll let it go through. From my POV, we are going to call it from Java thus we should not allow something that is different from native Java.

Reply:
I approved it because all I have left at nits. Things that you can fix if you want to, but are not important enough to me to hold up the PR over. If you don't want to make the change, then I am fine checking it in as is.